Update peer dependency to material-ui v3

jeanblanchard opened the pull request:
MUI has bumped their major version, which introduces a peer-dependency warning.

I opted to require v3 directly, instead of v1 or v3, since I see no good reason to continue using MUI v1.

leMaik commented:
Thanks for your first contribution! :tada: 
This library works fine with Material-UI 1.0.0, though and we shouldn't require users to update for no reason. Would you mind changing it to `^1.0.0 || ^2.0.0 || ^3.0.0`?

jeanblanchard commented:
Updated to `^1.0.0 || ^3.0.0` since MUI went directly from 1.5 to 3.0.
Also I initially messed up, and updated only the devDependency, not the peerDependency.
